vendors.json

```diff
@@ -132,7 +141,7 @@
   {
     "name": "Jenkins",
     "constant": "JENKINS",
-    "env": ["JENKINS_URL", "BUILD_ID"],
+    "env": { "any": ["JENKINS_URL", "BUILD_ID"] },
```

Comment:
Without JENKINS_URL, BUILD_ID is too generic might result in false positives. Thoughts?

Reply:
Yes. I was about to suggest the same. This is when we skip identifying vendors. We can just detect CI and not vendor.

Reply:
I do agree with the concerns that `BUILD_ID` might be too generic — what about adding it to this list?

https://github.com/watson/ci-info/blob/af36ac7d601e36719c15d1ba44df7af404493252/index.js#L56-L66

Reply:
Cloudbees (which is a popular enterprise Jenkins solution) appears to use `BUILD_ID` but not `BUILD_NUMBER`: https://docs.cloudbees.com/docs/admin-resources/latest/automating-with-jenkinsfile/working-with-the-env

Reply:
Lets have both BUILD_ID and BUILD_NUMBER for detecting `CI`. Should be good enough for most cases I guess

Reply:
I have changed my "jenkins" commit to do just this, also reverting the change to vendors.json.